components/brave_wallet/browser/wallet_data_files_installer.cc

```diff
@@ -122,16 +152,32 @@ ChainList ChainListReady(const base::FilePath& install_dir) {
 
   if (absolute_install_dir.empty()) {
     LOG(ERROR) << "Failed to get absolute install path.";
-    return chains;
   }
 
-  HandleParseChainList(absolute_install_dir, "chainlist.json", &chains);
-
-  return chains;
+  HandleParseTokenList(absolute_install_dir, "contract-map.json",
+                       mojom::CoinType::ETH,
+                       base::BindOnce(&UpdateTokenRegistry));
```

Comment:
We don't actually need to do this via callback but can call it directly in OnSanitizedTokenList only if it succeeds, which is the last line of that function.
This will make things easier with all those redundant updates we currently call because we always need to run the callback even if it fails. Also we were missing running callback in HandleParseTokenList when it has error.
All of it seems really unnecessary, same applies to chain list.
And we can move what's in UpdateChainListRegistry and UpdateTokenRegistry directly to the last line of `OnSanitized*`.